forti_vuln/forti_vuln.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+import socket
+import ssl
+from typing import Tuple
+
+from artemis.binds import Device, TaskStatus, TaskType
+from artemis.config import Config
+from artemis.module_base import ArtemisBase
+from artemis.task_utils import get_target_host
+from artemis.utils import throttle_request
+from karton.core import Task
+
+CONTROL_REQUEST = """POST /remote/VULNCHECK HTTP/1.1\r
+Host: {}\r
+User-Agent: {}\r
+Transfer-Encoding: chunked\r
+\r
+0\r
+\r
+\r
+"""
+
+VULN_CHECK_REQUEST = """POST /remote/VULNCHECK HTTP/1.1\r
+Host: {}\r
+User-Agent: {}\r
+Transfer-Encoding: chunked\r
+\r
+0000000000000000FF\r
+\r
+"""
+
+
+class FortiVuln(ArtemisBase):  # type: ignore[misc]
+    """
+    Checks FortiOS instance for CVE-2024-21762 vulnerability
+    source: https://github.com/BishopFox/CVE-2024-21762-check
+    """
+
+    identity = "forti_vuln"
+    filters = [
+        {"type": TaskType.DEVICE.value, "device": Device.FORTIOS.value},
+    ]
+
+    @staticmethod
+    def _send_req(context: ssl.SSLContext, address: Tuple[str, int], req: bytes) -> int:
+        try:
+            s = socket.create_connection(address, timeout=5)
+        except Exception:
+            return -1
+        ss = context.wrap_socket(s)
+        ss.send(req)
+        try:
+            ss.read(2048)
+            return 1
+        except socket.timeout:
+            return 0
+
+    def vuln_check(self, host: str, port: int) -> int:
+        context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+        context.check_hostname = False
+        context.verify_mode = ssl.CERT_NONE
+
+        http_host = f"{host}:{port}"
+        self.log.info(f"forti vuln scanning {http_host}")
+
+        user_agent = ""
```

Comment:
can we send an empty one? or sth generic, github link?

Reply:
added default user agent just in case